test/Tests/src/Data/Time/Date_Spec.enso

```diff
@@ -139,6 +139,35 @@ spec_with name create_new_date parse_date =
             date_1>=datetime . should_fail_with Type_Error_Data
             date_1==datetime . should_be_false
 
+        Test.specify "should create date before epoch start" <|
+            (create_new_date 100 1 2).year   . should_equal 100
+            (create_new_date 100 1 2).month  . should_equal 1
+            (create_new_date 100 1 2).day    . should_equal 2
+
+        Test.specify "Gregorian calendar related functionality should produce warning before epoch start" <|
+            expect_warning value =
+                ((Warning.get_all value).length > 0) . should_be_true
```

Comment:
I think it is worth checking here that the warning received is correct, not something else happening randomly.

This will be easier once the warning is wrapped in some atom - then you can check if `Warning.get_all value . map .value . exists (_.is_a Time_Error_Data)` or something like that.